lua/neotest/types/tree.lua

```diff
@@ -144,6 +144,21 @@ function Tree:iter_parents()
   end
 end
 
+function Tree:closest_node_with(data_attr)
```

Comment:
Let's add a small docstring describing what this is doing, because closest could be taken to include siblings.